src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -143,6 +105,17 @@ protected override bool ReleaseHandle()
                 Interop.Sys.Unlink(_path); // ignore errors; it's valid that the path may no longer exist
             }
 
+            // When the SafeFileHandle was opened, we likely issued an flock on the created descriptor in order to add
+            // an advisory lock.  This lock should be removed via closing the file descriptor, but close can be
+            // interrupted, and we don't retry closes.  As such, we could end up leaving the file locked,
+            // which could prevent subsequent usage of the file until this process dies.  To avoid that, we proactively
+            // try to release the lock before we close the handle.
+            if (_isLocked)
+            {
+                Interop.Sys.FLock(handle, Interop.Sys.LockOperations.LOCK_UN); // ignore any errors
```

Comment:
Even when close fails, we should no longer use the file descriptor.

I think we may assume `close` implies `flock(LOCK_UN)` and leave it out completely.

Reply:
If we want to do this, I suggest we do it in a separate PR.

Reply:
We mustn't do because the handle close may be not deterministic in presence of forked child processes (similar to https://github.com/dotnet/runtime/issues/58964).